### ISSUE TITLE
[Enhancement]: 

### ISSUE THREAD

**Issue by eyalya** [hashicorp/terraform-provider-aws#28952]
### Description

I would like to be able to create a default prefix or suffix to all resource naming using the provider in all key elements that cannot be duplicated. 
For example if I try to create an IAM role with the same name it will fail. I would like to add the prefix or suffix automatically using the provider, same as the default tags option. 

use case: 
I am working on a terraform code that has been built to be deployed on different account and now I need to change all name in order to deploy in another region. 

### Would you like to implement a fix?

None

**Comment by lvthillo** [comment]
IMO, it seems not really as a provider issue. You can use `locals` to create names in a dynamic way (e.g. based on region).

variables.tf

```
variable "region" {
  description = "Infrastructure region"
  type        = string
  default     = "eu-west-1"
}

variable "project_name" {
  description = "Name of project"
  type        = string
  default     = "test"
}
```

main.tf
```
provider "aws" {
  region = var.region
}

locals {
  name = "${var.project_name}-${data.aws_region.current.name}"
}

data "aws_region" "current" {}

resource "aws_sqs_queue" "terraform_queue" {
  name = "${local.name}-queue"
}
```

Plan:
```
  # aws_sqs_queue.terraform_queue will be created
  + resource "aws_sqs_queue" "terraform_queue" {
      ...
      + name                              = "test-eu-west-1-queue"
      ...
```

**Comment by eyalya** [comment]
Yes I know I can do that myself, the same as I can set tags myself, but there was a logic to add default tags on the provider level. It really helps, same as I think it will help here.
